retrieval_service/datastore/providers/cloudsql_mysql.py

```diff
@@ -0,0 +1,654 @@
+        res = await loop.run_in_executor(None, self.get_flight_sync, flight_id)
+        return res
+        
+    def search_flights_by_number_sync(
+        self,
+        airline: str,
+        number: str,
+    ) -> list[models.Flight]:
+        with self.__pool.connect() as conn:
+            s = text(
+                """
+                SELECT * FROM flights
+                  WHERE airline = :airline
+                  AND flight_number = :number
+                  LIMIT 10
+                """
+            )
+            params = {
+                "airline": airline,
+                "number": number,
+            }
+            results = (conn.execute(s, parameters=params)).mappings().fetchall()
+
+        res = [models.Flight.model_validate(r) for r in results]
+        return res
+    
+    async def search_flights_by_number(
+        self,
+        airline: str,
+        number: str,
+    ) -> list[models.Flight]:
+        loop = asyncio.get_running_loop()
+        res = await loop.run_in_executor(None, self.search_flights_by_number_sync, airline, number)
+        return res
+    
+    def search_flights_by_airports_sync(
+        self,
+        date: str,
+        departure_airport: Optional[str] = None,
+        arrival_airport: Optional[str] = None,
+    ) -> list[models.Flight]:
+        with self.__pool.connect() as conn:
+            s = text(
+                """
+                SELECT * FROM flights
+                  WHERE (CAST(:departure_airport AS CHAR(255)) IS NULL OR LOWER(departure_airport) LIKE LOWER(:departure_airport))
+                  AND (CAST(:arrival_airport AS CHAR(255)) IS NULL OR LOWER(arrival_airport) LIKE LOWER(:arrival_airport))
+                  AND departure_time >= CAST(:datetime AS DATETIME)
+                  AND (departure_time < DATE_ADD(CAST(:datetime AS DATETIME), interval 1 day))
+                  LIMIT 10
+                """
+            )
+            params = {
+                "departure_airport": departure_airport,
+                "arrival_airport": arrival_airport,
+                "datetime": datetime.strptime(date, "%Y-%m-%d"),
+            }
+
+            results = (conn.execute(s, parameters=params)).mappings().fetchall()
+
+        res = [models.Flight.model_validate(r) for r in results]
+        return res
+    
+    async def search_flights_by_airports(
+        self,
+        date: str,
+        departure_airport: Optional[str] = None,
+        arrival_airport: Optional[str] = None,
+    ) -> list[models.Flight]:
+        loop = asyncio.get_running_loop()
+        res = await loop.run_in_executor(None, self.search_flights_by_airports_sync, date, departure_airport, arrival_airport)
+        return res 
+    
+    def validate_ticket_sync(
+        self,
+        airline: str,
+        flight_number: str,
+        departure_airport: str,
+        departure_time: str,
+    ) -> Optional[models.Flight]:
+        raise NotImplementedError("Not Implemented")
```

Comment:
Can you add the implementation for these? You can refer to the alloydb.py for implementation. We have not yet update the cloudsql_postgres ones but these will be good to have so user can test out all the endpoints that are available.

Reply:
done